src/main/java/com/google/edith/servlets/Item.java

```diff
@@ -1,17 +1,50 @@
 package com.google.edith.servlets;
 
+import com.google.auto.value.AutoValue;
+
 /** Used to hold the properties of an item Entity in datastore. */
-public final class Item {
-  // TODO (malachibre): Add name, user, category, and receiptId fields.
-  // TODO (malachibre): Create a builder for this class inclduing checks
-  //                    for valid assignments.
-  double price;
-  long quantity;
-  String date;
-
-  public Item(double price, long quantity, String date) {
-    this.price = price;
-    this.quantity = quantity;
-    this.date = date;
+@AutoValue
+public abstract class Item {
+  public abstract String name();
+
+  public abstract String userId();
+
+  public abstract String category();
+
+  public abstract double price();
+
+  public abstract long quantity();
+
+  public abstract String date();
+
+  public abstract String expiration();
+
+  /**
+   * Creates an AutoValue Builder for this class.
+   *
+   * @return AutoValue Builder
+   */
+  public static Builder builder() {
+    return new AutoValue_Item.Builder();
+  }
+
+  /** Builder used to set the fields of this class. */
+  @AutoValue.Builder
+  public abstract static class Builder {
+    public abstract Builder setName(String value);
+
+    public abstract Builder setUserId(String value);
+
+    public abstract Builder setCategory(String value);
+
+    public abstract Builder setPrice(double value);
+
+    public abstract Builder setQuantity(long value);
+
+    public abstract Builder setDate(String value);
+
+    public abstract Builder setExpiration(String value);
+
```

Comment:
question: is this the final version of the Item file? like your code and Kai's code are both using this one currently

Reply:
Yeah, at least the class fields must be.